### PR TITLE
Adding context menu on the sticky scroll line (Fixes #162330)

### LINKED ISSUE #162330: Add sticky scroll context menu
It would be valuable to have a context menu for sticky scroll that allows to toggle it. We could also show it for breadcrumbs and also have the respective toggle in it.

### PR DESCRIPTION
**Pull request by aiday-mar**
Fixes https://github.com/microsoft/vscode/issues/162330.

Adding the possibility to toggle the sticky scroll line by right clicking on it. I had to modify the ClickLinkGesture in order to introduce a new parameter 'hasRightClick' which will differentiate between a normal click and a right click. Here is an example of how it currently works:

sticky-scroll-contextmenu

### CODE REVIEW

**Review by Tirednfedup — CHANGES_REQUESTED** [review]
Is there any suggestions?

File: src/vs/platform/actions/common/actions.ts [hunk 1]
```diff
@@ -103,6 +103,7 @@ export class MenuId {
 	static readonly SearchContext = new MenuId('SearchContext');
 	static readonly StatusBarWindowIndicatorMenu = new MenuId('StatusBarWindowIndicatorMenu');
 	static readonly StatusBarRemoteIndicatorMenu = new MenuId('StatusBarRemoteIndicatorMenu');
+	static readonly StickyScroll = new MenuId('StickyScrollMenu');
```
Comment (jrieken) [inline src/vs/platform/actions/common/actions.ts:106]: nit: use the `Context` suffix, eg.`StickyScrollContext`

**Review by jrieken — COMMENTED** [review]
This looks really good, left a tiny nit about the name of the menu (consistency matters)

I would also suggest to add the 'Toggle Breadcrumbs' command into the command. I believe sticky scroll and breadcrumbs are closely related and configured them in a single place sounds like a good idea. You can find the command here: [ToggleBreadcrumb](https://github.com/microsoft/vscode/blob/20e2a1524df3b40b3d96e42d468ac6d3ed23e545/src/vs/workbench/browser/parts/editor/breadcrumbsControl.ts#L500)

**Comment by gjsjohnmurray** [comment]
> I would also suggest to add the 'Toggle Breadcrumbs' command

Doing this would also act as a clue about where I might be able to go to toggle Sticky Scroll back on again. But only if right-click on breadcrumbs also gave this kind of menu. Without that I have to realize that the View menu is the place to go, or Command Palette.

Also, I wonder if the "Toggle prefix is necessary? And should other editor-view-related toggles appear here as well? For example Minimap (currently on a context menu of an editor's vertical scrollbar, as well as one from the minimap itself when visible).

**Comment by aiday-mar** [comment]
> > I would also suggest to add the 'Toggle Breadcrumbs' command
> 
> Doing this would also act as a clue about where I might be able to go to toggle Sticky Scroll back on again. But only if right-click on breadcrumbs also gave this kind of menu. Without that I have to realize that the View menu is the place to go, or Command Palette.
> 
> Also, I wonder if the "Toggle prefix is necessary? And should other editor-view-related toggles appear here as well? For example Minimap (currently on a context menu of an editor's vertical scrollbar, as well as one from the minimap itself when visible).

I would agree that having the menu appear also on the breadcrumbs bar would be more intuitive.

**Comment by jrieken** [comment]
> Also, I wonder if the "Toggle prefix is necessary? And should other editor-view-related toggles appear here as well?

That's a good point. It seems that we don't have a good story for this, we often use "Toggle XYZ" because of places that don't render the check mark, e.g the command palette. We should maybe expand on the existing toggled title so that the default label is "XYZ" (which gets rendered with the check mark_ and then have on/off labels like "Show XYZ" and "Hide XYZ" for places that don't renderer the check mark.

This is an interesting discussion but I am in favour of leaving this out of this PR.

**Review by aeschli — APPROVED** [review]

**Comment by jrieken** [comment]
There is https://github.com/microsoft/vscode/pull/162672 for the toogle-title disuccsion

**Pull request merged**